impl/authorization/authorization.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package authorization contains the authorization module implementation.
+package authorization
+
+import (
+	"fmt"
+
+	"github.com/google/keytransparency/core/authentication"
+	"github.com/google/keytransparency/core/authorization"
+
+	authzpb "github.com/google/keytransparency/core/proto/authorization"
+)
+
+type authz struct {
+	policy *authzpb.AuthorizationPolicy
+}
+
+// New creates a new instance of the authorization module.
+func New() authorization.Authorization {
+	return &authz{}
+}
+
+// IsAuthorized verifies that the identity issuing the call (from ctx) is
+// authorized to carry the given permission. A call is authorized if:
+//  1. userID matches the identity in sctx,
+//  2. or, sctx's identity is authorized to do the action in mapID and appID.
+func (a *authz) IsAuthorized(sctx *authentication.SecurityContext, mapID, appID int64,
+	userID string, permission authzpb.Permission) error {
+	// Case 1.
+	if sctx.Identity() == userID {
+		return nil
+	}
+
+	// Case 2.
+	rLabel := resourceLabel(mapID, appID)
+	roles, ok := a.policy.GetResourceToRoles()[rLabel]
+	if !ok {
+		return fmt.Errorf("resource defined by <mapID=%v, appID=%v> does not exist", mapID, appID)
```

Comment:
does not have a defined policy?

Should this be access denied? If a resource doesn't have a policy it should failed closed.